### PR TITLE
Update CIME to ESMCI cime5.7.6

### PR DESCRIPTION
**Pull request by jgfouca**
Update CIME to ESMCI cime5.7.6

Squash merge of jgfouca/branch-for-to-acme-2019-01-14

Features:
* Improve error/exception handling throughout CIME
* Update pio1
* Update pio2
* Improve errors messages coming from generic_xml
* Remove support for CLM4.0
* bring in latest nuopc developments

Bug fixes:
* docn: make sure optional arg exists before using
* improve the logic of the continue run check
* case.cmpgen_namelists needs to be able to write to the case
* Fix Case context manager enforcement

[BFB]

### CODE REVIEW

File: cime/scripts/Tools/Makefile [hunk 1]
```diff
@@ -105,7 +105,9 @@ endif
 CPPDEFS := $(USER_CPPDEFS) -D$(OS)
 
 # Unless DEBUG mode is enabled, use NDEBUG to turn off assert statements.
-ifneq ($(strip $(DEBUG)),TRUE)
+ifeq ($(strip $(DEBUG)),TRUE)
+   CPPDEFS += -DDEBUG
```
[the remark this answers is not in the document]
Comment (rljacob) [reply cime/scripts/Tools/Makefile]: No I think they updates FATES to a version that doesn't use DEBUG as a fortran variable.

**Comment by jgfouca** [comment]
Everything looks what I expected. Testing now.

File: cime/scripts/Tools/Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (jgfouca) [reply cime/scripts/Tools/Makefile]: Still getting DEBUG errors unfortunately.

[the remark this answers is not in the document]
Comment (jgfouca) [reply cime/scripts/Tools/Makefile]: Nevermind, just needed to update subs.

[the remark this answers is not in the document]
Comment (jgfouca) [reply cime/scripts/Tools/Makefile]: Some tests still failing due to this, needs to be removed for now.

**Comment by jgfouca** [comment]
Squashed to next.

**Comment by rljacob** [comment]
Was this exactly cime5.7.6 or did you pull from the head of master?   Because that was just tagged as cime5.7.7.

**Comment by jgfouca** [comment]
I pulled head from a couple days ago. 5.7.6 was the most recent tag at the time.

**Pull request merged**